test/functional/apps/dashboard_elements/links/links_create_edit.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const testSubjects = getService('testSubjects');
+  const dashboardAddPanel = getService('dashboardAddPanel');
+  const deployment = getService('deployment');
+  const dashboardPanelActions = getService('dashboardPanelActions');
+  const { dashboardLinks, dashboard, common, header } = getPageObjects([
+    'dashboardLinks',
+    'dashboard',
+    'common',
+    'header',
+  ]);
+
+  async function createSomeLinks() {
+    await dashboardLinks.addExternalLink(
+      `${deployment.getHostPort()}/app/foo`,
+      true,
+      true,
+      'Link to new tab'
+    );
+    await dashboardLinks.addExternalLink(`${deployment.getHostPort()}/app/bar`, false, true);
+
+    await dashboardLinks.addDashboardLink(DASHBOARD_NAME);
+    await dashboardLinks.addDashboardLink('links 001');
+  }
+
+  const DASHBOARD_NAME = 'Test Links panel';
+  const LINKS_PANEL_NAME = 'Some links';
+
+  describe('links panel creation', async () => {
+    before(async () => {
+      await dashboard.navigateToApp();
+      await dashboard.preserveCrossAppState();
+      await dashboard.gotoDashboardLandingPage();
+      await dashboard.clickNewDashboard();
+      await dashboard.saveDashboard(DASHBOARD_NAME, { exitFromEditMode: false });
+      await dashboard.loadSavedDashboard(DASHBOARD_NAME);
+      await dashboard.switchToEditMode();
+    });
+
+    it('can not add an external link that violates externalLinks.policy', async () => {
+      await dashboardAddPanel.clickEditorMenuButton();
+      await dashboardAddPanel.clickAddNewEmbeddableLink('links');
+
+      await dashboardLinks.setExternalUrlInput('https://danger.example.com');
+      expect(await testSubjects.exists('links--linkDestination--error')).to.be(true);
+      await dashboardLinks.clickLinkEditorCloseButton();
+      await dashboardLinks.clickPanelEditorCloseButton();
+    });
```

Comment:
**Note for later:** I wonder if we could make this a component unit test? Not something to do now, though

Reply:
We'd need to mock the kibana config. Might be possible to convert this to a unit test in the future.